services/rollout-service/pkg/versions/versions.go

```diff
@@ -218,21 +234,53 @@ func (v *versionClient) ConsumeEvents(ctx context.Context, processor VersionEven
 							continue
 						}
 
-						processor.ProcessKuberpultEvent(ctx, KuberpultEvent{
-							Application:      app.Name,
-							Environment:      env.Name,
-							EnvironmentGroup: envGroup.EnvironmentGroupName,
-							Team:             tm,
-							IsProduction:     env.Priority == api.Priority_PROD,
-							Version: &VersionInfo{
-								Version:        app.Version,
-								SourceCommitId: sc,
-								DeployedAt:     dt,
-							},
-						})
+						if v.manageArgoAppsEnabled && len(v.manageArgoAppsFilter) > 0 && strings.Contains(v.manageArgoAppsFilter, app.Name) {
+							manifestPath := path.Join("environments", env.Name, "applications", app.Name, "manifests", "manifests.yaml")
```

Comment:
You have already most of the code that generates the applications in rollout-service/pkg/argo. I would move this code there as well for simplicity.